ENIGMAsystem/SHELL/Graphics_Systems/General/GSstdraw.cpp

```diff
@@ -421,94 +422,68 @@ void draw_roundrect(gs_scalar x1, gs_scalar y1,gs_scalar x2, gs_scalar y2, float
-    draw_primitive_begin(pr_linestrip);
-  }else{
-    draw_primitive_begin(pr_trianglefan);
-  }
-  //Draw left 1/4 circle
-  for (gs_scalar i = M_PI; i > M_PI_2; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex(x1+rad+xc1, y1+rad+yc1);
-  }
-  draw_vertex(x1+rad, y1);
-  //Draw right 1/4 circle
-  for (gs_scalar i = M_PI_2; i > 0; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex(x2-rad+xc1, y1+rad+yc1);
-  }
-  draw_vertex(x2, y1+rad);
-  //Bottom right 1/4 circle
-  for (gs_scalar i = 2 * M_PI; i > M_PI+M_PI_2; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex(x2-rad+xc1, y2-rad+yc1);
-  }
-  draw_vertex(x2-rad, y2);
-  //Bottom left 1/4 circle
-  for (gs_scalar i = M_PI+M_PI_2; i > M_PI; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex(x1+rad+xc1, y2-rad+yc1);
-  }
-  draw_vertex(x1, y2-rad);
-  if (outline) {
-    draw_vertex(x1, y1+rad);
+      draw_vertex(x2,y2-rad);
   }
   draw_primitive_end();
 }
 
 void draw_roundrect_color(gs_scalar x1, gs_scalar y1, gs_scalar x2, gs_scalar y2, float rad, int col1, int col2, bool outline) {
+  gs_scalar alpha = draw_get_alpha();
   if (x1>x2) {
     gs_scalar t=x2;
     x2=x1;
     x1=t;
-  }
+    }
   if (y1>y2) {
     gs_scalar t=y2;
     y2=y1;
     y1=t;
   }
-  x1++, y1++; //This fixes an off-by-one error when drawing over a regular draw_rectangle
-  gs_scalar alpha = draw_get_alpha();
+
   gs_scalar pr = 2 * M_PI / enigma::circleprecision;
   if (outline) {
     draw_primitive_begin(pr_linestrip);
-  }else{
+  } else {
     draw_primitive_begin(pr_trianglefan);
-    draw_vertex_color(x1+(x2-x1)/2.0, y1+(y2-y1)/2.0, col1, alpha);
-  }
-  //Draw left 1/4 circle
-  for (gs_scalar i = M_PI; i > M_PI_2; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex_color(x1+rad+xc1, y1+rad+yc1, col2, alpha);
-  }
-  draw_vertex_color(x1+rad, y1, col2, alpha);
-  //Draw right 1/4 circle
-  for (gs_scalar i = M_PI_2; i > 0; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex_color(x2-rad+xc1, y1+rad+yc1, col2, alpha);
-  }
-  draw_vertex_color(x2, y1+rad, col2, alpha);
-  //Bottom right 1/4 circle
-  for (gs_scalar i = 2 * M_PI; i > M_PI+M_PI_2; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex_color(x2-rad+xc1, y2-rad+yc1, col2, alpha);
-  }
-  draw_vertex_color(x2-rad, y2, col2, alpha);
-  //Bottom left 1/4 circle
-  for (gs_scalar i = M_PI+M_PI_2; i > M_PI; i -= pr) {
-    gs_scalar xc1=cos(i)*rad,yc1=-sin(i)*rad;
-    draw_vertex_color(x1+rad+xc1, y2-rad+yc1, col2, alpha);
-  }
-  draw_vertex_color(x1, y2-rad, col2, alpha);
-  draw_vertex_color(x1, y1+rad, col2, alpha);
+    draw_vertex_color((x2+x1)/2,(y2+y1)/2, col1, alpha);
+  }
+
+  for (gs_scalar i = 0; i <= 2*M_PI*0.25; i += pr)
+    draw_vertex_color((x2+rad*cos(i))-rad,(y2+rad*sin(i))-rad, col2, alpha);
+
+  for (gs_scalar i = 2*M_PI*0.24; i <= 2*M_PI*0.50; i+= pr)
```

Comment:
Oh lord... shouldn't you be drawing the corners manually rather than computing `sin(0)`, etc? Also, @TheExDeus, I have already added those constants to a unified place included at the top of this file (`Universal_System/math_consts.h`).